base/libc.jl

```diff
@@ -382,9 +382,15 @@ Interface to the C `rand()` function. If `T` is provided, generate a value of ty
 by composing two calls to `rand()`. `T` can be `UInt32` or `Float64`.
 """
 rand() = ccall(:rand, Cint, ())
-# RAND_MAX at least 2^15-1 in theory, but we assume 2^16-1 (in practice, it's 2^31-1)
-rand(::Type{UInt32}) = ((rand() % UInt32) << 16) ⊻ (rand() % UInt32)
-rand(::Type{Float64}) = rand(UInt32) / 2^32
+@static if Sys.iswindows()
+    # Windows RAND_MAX is 2^15-1
+    rand(::Type{UInt32}) = ((rand() % UInt32) << 17) ⊻ ((rand() % UInt32) << 8) ⊻ (rand() % UInt32)
+else
+    # RAND_MAX is at least 2^15-1 in theory, but we assume 2^16-1
+    # on non-Windows systems (in practice, it's 2^31-1)
+    rand(::Type{UInt32}) = ((rand() % UInt32) << 16) ⊻ (rand() % UInt32)
+end
+rand(::Type{Float64}) = rand(UInt32) * 2.0^-32
```

Comment:
Use `ldexp`?

Reply:
`@code_native g(UInt32(3))` reports an enormous amount of code for `g(x) = ldexp(Float64(x), -32)` for some reason…

Reply:
I think `ldexp` is probably slower here here because it doesn't seem to constant-propagate the exponent?